services/app-web/src/pages/StateSubmission/RateDetails/RateDetailsSchema.ts

```diff
@@ -10,171 +10,176 @@ import {
+                        .typeError('The end date must be in MM/DD/YYYY format')
+                        .min(
+                            Yup.ref('effectiveDateStart'),
+                            'The end date must come after the start date'
+                        ),
+                }),
+                actuaryContacts: Yup.array().of(
+                    Yup.object().shape({
+                        name: Yup.string().required('You must provide a name'),
+                        titleRole: Yup.string().required(
+                            'You must provide a title/role'
+                        ),
+                        email: Yup.string()
+                            .email('You must enter a valid email address')
+                            .required('You must provide an email address'),
+                        actuarialFirm: Yup.string()
+                            .required('You must select an actuarial firm')
+                            .nullable(),
+                        actuarialFirmOther: Yup.string()
+                            .when('actuarialFirm', {
+                                is: 'OTHER',
+                                then: Yup.string()
+                                    .required('You must enter a description')
+                                    .nullable(),
+                            })
+                            .nullable(),
+                    })
+                ),
+                addtlActuaryContacts: Yup.array().of(
+                    Yup.object().shape({
+                        name: Yup.string().required('You must provide a name'),
+                        titleRole: Yup.string().required(
+                            'You must provide a title/role'
+                        ),
+                        email: Yup.string()
+                            .email('You must enter a valid email address')
+                            .required('You must provide an email address'),
+                        actuarialFirm: Yup.string()
+                            .required('You must select an actuarial firm')
+                            .nullable(),
+                        actuarialFirmOther: Yup.string()
+                            .when('actuarialFirm', {
+                                is: 'OTHER',
+                                then: Yup.string()
+                                    .required('You must enter a description')
+                                    .nullable(),
+                            })
+                            .nullable(),
+                    })
+                ),
+                actuaryCommunicationPreference: Yup.string().required(
+                    'You must select a communication preference'
+                )
+            })
     })
 const RateDetailsFormSchema = (activeFeatureFlags?: FeatureFlagSettings) => {
-    return activeFeatureFlags?.['rate-edit-unlock'] ||  activeFeatureFlags?.['link-rates'] ?
-    Yup.object().shape({
```

Comment:
This function needs to return the `Yup.object`. 

Also, validations are not working because RateDetailsV2 expects `rateForms` and RateDetails(V1) expects `rateInfos`. Whichever one we want this to be, either V1 or V2 has to be refactored to use it.

Reply:
Also, failing `rateDetailsSchema.test.ts` tests were due returning `rateInfo` instead of `rateForms`.